### PR TITLE
Add regression test for STI duplicate foreign key deduplication

### PR DESCRIPTION
**Pull request by janedbal**
When multiple STI child entities have a `ManyToOne` relation to the same target entity using the same join column, `SchemaTool` called `addForeignKeyConstraint` twice with the same constraint name, triggering a DBAL deprecation ("Overwriting an existing foreign key constraint").

The fix skips the redundant `addForeignKeyConstraint` call when an identical FK (same local columns, same foreign table, same foreign columns) was already added.

See https://github.com/doctrine/dbal/pull/7125

---

**EDIT (after rebase):** The production fix has since landed in `3.6.x` via #12477 ("Avoid adding the same foreign key twice for STI"), which implements the same deduplication more thoroughly (it also compares FK options). Rebasing dropped the now-redundant `SchemaTool` change, so this PR now **only adds the regression test** — upstream merged the fix without one, and this test guards the STI same-join-column scenario going forward.

### CODE REVIEW

**Comment by janedbal** [comment]
Any chance somebody looks into this one?

**Comment by greg0ire** [comment]
It seems there are CI jobs failing. Please take a look at [this guide](https://www.doctrine-project.org/contribute/#dealing-with-checks-and-tools) for more on how to handle those.

**Comment by janedbal** [comment]
@greg0ire None of those were related to our changes. I just rebased to trigger new pipeline.

**Comment by greg0ire** [comment]
That's cool. Bear in mind that I will always prioritize reviewing PRs with a green build.

File: tests/Tests/ORM/Tools/SchemaToolTest.php [hunk 1]
```diff
@@ -523,6 +527,39 @@ private static function columnIsIndexed(DbalTable $table, string $column): bool
...
+        if (class_exists(ForeignKeyConstraintEditor::class)) {
+            self::assertSame('sti_box', $foreignKey->getReferencedTableName()->toString());
+            self::assertSame(['box_id'], array_map(
+                static fn (UnqualifiedName $name) => $name->toString(),
+                $foreignKey->getReferencingColumnNames(),
+            ));
+        } else {
+            self::assertSame('sti_box', $foreignKey->getForeignTableName());
+            self::assertSame(['box_id'], $foreignKey->getLocalColumns());
+        }
```
Comment (greg0ire) [inline tests/Tests/ORM/Tools/SchemaToolTest.php:615]: Please use an early exit, making sure the happy path (recent DBAL) has the lowest indention level.

Comment (janedbal) [reply tests/Tests/ORM/Tools/SchemaToolTest.php:615]: Fixed

File: tests/Tests/ORM/Tools/SchemaToolTest.php [hunk 2]
```diff
@@ -581,6 +585,41 @@ private static function columnIsIndexed(DbalTable $table, string $column): bool
 
         return false;
     }
+
+    public function testSingleTableInheritanceWithSameForeignKeyInChildrenDoesNotTriggerDeprecation(): void
```
Comment (greg0ire) [inline tests/Tests/ORM/Tools/SchemaToolTest.php:589]: Should there be a call to https://github.com/doctrine/deprecations/blob/5e059379444d9e38df997f4542c9d386239e27b6/src/PHPUnit/VerifyDeprecations.php#L26?
